File: RB120_Object_Oriented_Programming/Lesson2/classes_and_objects_1/practice1.py
class Person:
    def __init__(self, name):
        parsed_name = name.split()
        self.first_name = parsed_name[0]
        self.last_name = parsed_name[1] if len(parsed_name) > 1 else ''


    @property
    def name(self):
        return (self.first_name + ' ' + self.last_name).strip()

    @name.setter
    def name(self, input_name):
        parsed_name = input_name.split()
        self.first_name = parsed_name[0]
        self.last_name = parsed_name[1] if len(parsed_name) > 1 else ''
        self._name = (self.first_name + ' ' + self.last_name).strip()

    def __str__(self):
        return self.name

bob = Person('Robert')
print(bob.name)
print(bob.first_name)
print(bob.last_name)
bob.last_name = 'Smith'
print(bob.name)

bob.name = "John Adams"
print(bob.first_name)
print(bob.last_name)
print(bob.name)

print(bob)

File: RB120_Object_Oriented_Programming/Lesson2/classes_and_objects_1/practice1.rb
class Person
  attr_reader :first_name, :last_name
  attr_writer :last_name

  def initialize(name)

    parsed_names = name.split
    @first_name = parsed_names.first

    if parsed_names.size > 1
      @last_name = parsed_names.last
    else
      @last_name = ''
    end

  end


  def name
    if @last_name.empty?
      @first_name
    else
      @first_name + ' ' + @last_name
    end
  end


end


bob = Person.new('Robert')
p bob.name
p bob.first_name
p bob.last_name
bob.last_name = 'Smith'
p bob.name
"hello".split